core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraph.scala

```diff
@@ -164,12 +164,20 @@ private[ui] object RDDOperationGraph extends Logging {
    *
    * For the complete DOT specification, see http://www.graphviz.org/Documentation/dotguide.pdf.
    */
-  def makeDotFile(graph: RDDOperationGraph): String = {
+  def makeDotFile(graph: RDDOperationGraph, stageId: String): String = {
     val dotFile = new StringBuilder
     dotFile.append("digraph G {\n")
-    dotFile.append(makeDotSubgraph(graph.rootCluster, indent = "  "))
-    graph.edges.foreach { edge => dotFile.append(s"""  ${edge.fromId}->${edge.toId};\n""") }
-    dotFile.append("}")
+    try {
+      dotFile.append(makeDotSubgraph(graph.rootCluster, indent = "  "))
+      graph.edges.foreach { edge => dotFile.append(s"""  ${edge.fromId}->${edge.toId};\n""") }
+      dotFile.append("}")
+    } catch {
+      case t: Throwable =>
```

Comment:
@XuTingJun, if an OOM happens then there's nothing we can do. In general we should not try to catch OOMs; instead we should prevent it in the first place if possible. Elsewhere in Spark we never catch `Throwable`, but only `NonFatal(e)`.